api/test/unit/utils/test_image_proxy.py

```diff
@@ -443,16 +442,17 @@ def test_photon_get_raises_by_not_allowed_types(image_type):
     "headers, expected_cache_val",
     [
         ({"Content-Type": "image/tiff"}, b"tiff"),
-        (None, b"unknown"),
+        ({"Content-Type": "unknown"}, b"unknown"),
     ],
 )
 def test_photon_get_saves_image_type_to_cache(redis, headers, expected_cache_val):
     image_url = TEST_IMAGE_URL.replace(".jpg", "")
     image = ImageFactory.create(url=image_url)
-
-    with mock.patch("requests.head") as mock_head, pytest.raises(UnsupportedMediaType):
-        mock_head.return_value.headers = headers
+    pook.on()
+    pook.head(image_url, reply=200, response_headers=headers)
+    with pytest.raises(UnsupportedMediaType):
         photon_get(image_url, image.identifier)
 
     key = f"media:{image.identifier}:thumb_type"
     assert redis.get(key) == expected_cache_val
+    pook.off()
```

Comment:
Please use the `pook.on` decorator for unit tests, otherwise `pook` won't get turned off properly after the test runs if any of the assertions or code-paths fail (causing pook to leak into other tests that might not expect it).

Although, IIRC, we can't use the decorator here because we need pook to avoid intercepting the Elasticsearch client's HTTP requests in `ImageFactory.create`. In that case, you can see some fixtures I wrote to handle this case for a PR that explored other changes: https://github.com/WordPress/openverse/blob/980ef96e0e16fc758f78be9f2aadf595f75948f6/api/test/unit/conftest.py#L144C1-L171

To use those fixtures, add then to `conftest.py` as they are in the linked commit and then add `pook_off` to this test and call `pook.on` when needed.

Alternatively, use `pook.use` as a context manager around `photon_get`: https://pook.readthedocs.io/en/latest/examples.html#context-manager-for-isolated-mocking

Reply:
Thanks for providing the detailed answer and good call on `pook` not getting turned off properly in case of failure. I'll make the updates soon :)